libs/deposits/src/lib/use-persistent-deposit.ts

```diff
@@ -0,0 +1,32 @@
+import { useLocalStorageSnapshot } from '@vegaprotocol/react-helpers';
+import { useCallback, useMemo } from 'react';
+
+const STORAGE_KEY = 'DEPOSIT_FORM_STORAGE';
+interface PersistedDeposit {
+  assetId: string;
+  amount?: string;
+}
+type PersistedDepositData = Record<string, PersistedDeposit>;
+
+export const usePersistentDeposit = (
+  assetId?: string
+): [PersistedDeposit, (entry: PersistedDeposit) => void] => {
+  const [value, saveValue] = useLocalStorageSnapshot(STORAGE_KEY);
+  const parsedValues = useMemo(
+    () => JSON.parse(value || '{}') as PersistedDepositData,
+    [value]
+  );
+  const savedData = useMemo(() => {
+    const index = assetId || Object.keys({ ...parsedValues }).pop() || '';
```

Comment:
suggestion (non blocking): This part seems a little odd to me. Can you add a comment or alternatively it may be better to let the hook return undefined if there isn't a matching deposit for the asset found.

Reply:
now it looks little different, but basically it just returns last added value in case when no key (assetId) is given